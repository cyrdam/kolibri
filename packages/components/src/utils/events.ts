// TODO: Should be synchronized with enums/events.ts
export type KoliBriEventType =
	| 'blur'
	| 'change'
	| 'changePage'
	| 'changePageSize'
	| 'click'
	| 'close'
	| 'focus'
	| 'input'
	| 'mousedown'
	| 'reset'
	| 'selection-change' // todo remove
	| 'selectionChange'
	| 'sort'
	| 'submit'
	| 'toggle';

export function stopPropagation(event: Event): void {
	event.stopImmediatePropagation();
	event.stopPropagation();
}

function createKoliBriEvent<T>(type: KoliBriEventType, detail: T): CustomEvent {
	const event = new CustomEvent(`kol-${type}`, {
		bubbles: true,
		cancelable: true,
		composed: true,
		detail: detail,
	});
	return event;
}

function dispatchKoliBriEvent<T>(target: EventTarget, type: KoliBriEventType, detail?: T): boolean {
	const dispatch = target.dispatchEvent(createKoliBriEvent(type, detail));
	return dispatch;
}

/**
 * @deprecated
 */
export function tryToDispatchKoliBriEvent<T>(type: KoliBriEventType, target?: EventTarget, detail?: T): void {
	target && dispatchKoliBriEvent(target, type, detail);
}

export function dispatchDomEvent(target: HTMLElement, type: KoliBriEventType) {
	target.dispatchEvent(new Event(type, { bubbles: true, composed: true }));
}

export function preventDefaultAndStopPropagation(event: Event) {
	event.preventDefault();
	stopPropagation(event);
}
