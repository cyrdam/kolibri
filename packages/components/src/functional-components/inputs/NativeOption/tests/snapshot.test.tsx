import { h } from '@stencil/core';
import NativeOptionFc from '../NativeOption';
import { renderFunctionalComponentToSpecPage } from '../../../../utils/testing';

describe('NativeOptionFc', () => {
	it('should render and match snapshot', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <NativeOptionFc value="testValue" label="testLabel" />);
		expect(page.root).toMatchSnapshot();
	});

	it('should render an option tag with the correct value and label', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <NativeOptionFc value="testValue" label="testLabel" />);
		const option = page.root;
		expect(option?.tagName).toBe('OPTION');
		expect(option?.getAttribute('value')).toBe('testValue');
		expect(option?.textContent).toBe('testLabel');
	});

	it('should set selected attribute correctly based on selectedValue', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <NativeOptionFc value="testValue" label="testLabel" selectedValue="testLabel" />);
		const option = page.root;
		expect(option?.hasAttribute('selected')).toBe(true);
	});

	it('should set selected attribute correctly based on selected prop', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <NativeOptionFc value="testValue" label="testLabel" selected />);
		const option = page.root;
		expect(option?.hasAttribute('selected')).toBe(true);
	});
});
