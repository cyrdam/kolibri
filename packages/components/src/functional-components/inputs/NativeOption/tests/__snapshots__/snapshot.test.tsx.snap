// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeOptionFc should render and match snapshot 1`] = `
<option value="testValue">
  testLabel
</option>
`;
